nav2_dwb_controller/nav_2d_utils/CMakeLists.txt

```diff
@@ -28,41 +28,64 @@ set(dependencies
   nav_2d_msgs
 )
 
-include_directories(
-    include
-)
-
 add_library(conversions SHARED
   src/conversions.cpp)
-
-ament_target_dependencies(conversions
-  ${dependencies}
+target_include_directories(conversions
+  PUBLIC
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+    "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+)
+target_link_libraries(conversions PUBLIC
+  ${geometry_msgs_TARGETS}
+  ${nav_2d_msgs_TARGETS}
+  ${nav_msgs_TARGETS}
+  rclcpp::rclcpp
+  tf2::tf2
+)
+target_link_libraries(conversions PRIVATE
+  ${tf2_geometry_msgs_TARGETS}
+  nav2_util::nav2_util_core
 )
 
 add_library(path_ops SHARED
   src/path_ops.cpp)
-
-ament_target_dependencies(path_ops
-  ${dependencies}
+target_include_directories(path_ops
+  PUBLIC
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+    "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+)
+target_link_libraries(path_ops PUBLIC
+  ${nav_2d_msgs_TARGETS}
+)
+target_link_libraries(path_ops PRIVATE
+  ${geometry_msgs_TARGETS}
 )
 
 add_library(tf_help SHARED
   src/tf_help.cpp
 )
-
-ament_target_dependencies(tf_help
-  ${dependencies}
+target_include_directories(tf_help
+  PUBLIC
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+    "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+)
+target_link_libraries(tf_help PUBLIC
+  tf2_ros::tf2_ros
+  ${geometry_msgs_TARGETS}
+  rclcpp::rclcpp
+  ${nav_2d_msgs_TARGETS}
+  tf2::tf2
+  ${tf2_geometry_msgs_TARGETS}
+  conversions
```

Comment:
Is there an alias target available or one that could be created to reduce the chances of linker errors due to typos on the library name?

Reply:
I've gotten ALIAS targets working well in about 120 packages. If you do the ALIAS as `${PROJECT_NAME}::your_target`, then the build target name is the same as ament's exported name. From what I've seen, ALIAS targets don't show up in the install interface, and naming them the same is recommended: https://youtu.be/bsXLMQ6WgIk?si=SPk624LzEa2UhLU5&t=3133

I have yet to have issues doing this method, and will routinely create ALIAS targets so my code in `test` can link to the right library with configure errors if there are typos.